src/libs/conduit/conduit_data_array.cpp

```diff
@@ -160,69 +161,51 @@ template <typename T>
 bool
 DataArray<T>::diff(const DataArray<T> &array, Node &info) const 
 { 
+    bool res = false;
     info.reset();
 
     index_t t_nelems = number_of_elements();
     index_t o_nelems = array.number_of_elements();
 
-    for(size_t i = 0; i < (size_t)std::min(t_nelems, o_nelems); i++)
+    DataType info_dtype(array.dtype().id(), std::max(t_nelems, o_nelems));
+    info.set(info_dtype);
+    T* info_ptr = (T*)info.data_ptr();
+
+    // TODO(JRC): This is a bit sloppy, but it feels like the only decent
+    // way to signal missing values in cases where the lengths are mismatched.
+    T t_fill, o_fill;
+    if(std::numeric_limits<T>::has_infinity)
```

Comment:
@cyrush: This is the biggest/most unusual change I can think of since some of the Travis jobs started failing. Do you think this is the culprit behind the failures, or do you suspect that the problem lies elsewhere? If we can't use the functionality in the "limits" header due to compatibility concerns, what do you think the best way to populate the extra entries in during a `DataArray<T>::diff` operation?

Reply:
The travis issues don't have anything to do with your changes.  Not sure what is going on in appveyor. Not sure about populating with infinity, sound idea, but might create headaches in practice.

Reply:
May be safer to pick one path to avoid surprises based on the compiler -- might have to be he non has_infinity path.